### PR TITLE
Add tests for variant analysis history item

### PR DESCRIPTION
**Pull request by norascheuch**
🚨 Please review this PR commit-by-commit.

We're repeating what we've done for the `RemoteQueryHistoryItem` tests for now.

Separately we'll think about setting up tests that check for both remote queries and variant analysis in the query history.

At the moment we'd like to focus on just adding some test coverage for `VariantAnalysisHistoryItem`.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Comment by elenatanasoiu** [comment]
Depends on these PRs getting merged in first:
- https://github.com/github/vscode-codeql/pull/1655
- https://github.com/github/vscode-codeql/pull/1654

**Comment by shati-patel** [comment]
(just closed and re-opened in an attempt to kick off PR checks!)

**Review by koesie10 — APPROVED** [review]
LGTM, I have some small improvement suggestions but this will definitely increase our test coverage

File: extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/variant-analysis-history.test.ts [hunk 1]
```diff
@@ -120,6 +120,31 @@ describe('Variant Analyses and QueryHistoryManager', function() {
...
+    await qhm.handleRemoveHistoryItem(qhm.treeDataProvider.allHistory[0]);
+
+    expect(removeVariantAnalysisStub).calledOnceWithExactly(rawQueryHistory[0].historyItemId);
+    expect(rehydrateVariantAnalysisStub).to.have.callCount(2);
+
+    expect(rehydrateVariantAnalysisStub.getCall(0).args[0]).to.deep.eq(rawQueryHistory[0].variantAnalysis);
+    expect(rehydrateVariantAnalysisStub.getCall(0).args[1]).to.deep.eq(rawQueryHistory[0].status);
+
+    expect(rehydrateVariantAnalysisStub.getCall(1).args[0]).to.deep.eq(rawQueryHistory[1].variantAnalysis);
+    expect(rehydrateVariantAnalysisStub.getCall(1).args[1]).to.deep.eq(rawQueryHistory[1].status);
```
[the remark this answers is not in the document]
Comment (elenatanasoiu) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/variant-analysis-history.test.ts]: Yeah they are kinda redundant. I've removed the extra checks.

File: extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/variant-analysis-history.test.ts [hunk 2]
```diff
@@ -120,6 +120,31 @@ describe('Variant Analyses and QueryHistoryManager', function() {
...
+
+    expect(openRemoteQueryResultsStub).calledOnceWithExactly(rawQueryHistory[1].historyItemId);
+    expect(qhm.treeDataProvider.allHistory).to.deep.eq(rawQueryHistory.slice(1));
+
+    // Add it back to the history
+    qhm.addQuery(rawQueryHistory[0]);
+    expect(removeVariantAnalysisStub).to.have.callCount(1);
+    expect(rehydrateVariantAnalysisStub).to.have.callCount(2);
```
[the remark this answers is not in the document]
Comment (elenatanasoiu) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/variant-analysis-history.test.ts:142]: We call `sandbox.restore()` in the `beforeEach()` block, so the count is reset for each of the tests. 

I've double checked this by adding these lines at the top of this test:
```
    expect(removeVariantAnalysisStub).to.have.callCount(0);
    expect(rehydrateVariantAnalysisStub).to.have.callCount(0);
```

They passed, which indicates the count starts from zero.

**Pull request merged**